### PR TITLE
TGEN: PR for master branch since working state has been achieved 

### PR DESCRIPTION
**Pull request by pemami4911**
The TGEN is in a functional state; current support is for geometric RRTstar and RRTConnect. In the params launch file, you can select whichever planner you'd like to use. So far, without obstacle avoidance, RRTConnect runs faster and with higher quality paths. Next implementation steps involves the module that will take PCL data and form a traversability map that the TGEN can use for state validation. See the README for details concerning the new `sub8_msgs::PathPlan` service. 

See the commits from today (January 31st) for code review purposes.  The main work was removing unnecessary code related to system dynamics and updating the remaining code. 

Minor change was made to the C++ alarm system; the data structure containing AlarmRaisers was changed to a std::map to allow indexing by the individual alarm names.

### CODE REVIEW

File: command/sub8_alarm/include/sub8_alarm/alarm_helpers.h [hunk 1]
```diff
@@ -3,13 +3,13 @@
  * Date: 11/9/2015
  */
 
-#ifndef ALARM_HELPERS_H_
-#define ALARM_HELPERS_H_
+#ifndef _ALARM_HELPERS_H
+#define _ALARM_HELPERS_H
```
Comment (jpanikulam) [inline command/sub8_alarm/include/sub8_alarm/alarm_helpers.h:7]: No-Change: You can use #pragma once to replace header guards

File: gnc/sub8_trajectory_generator/src/tgen_manager.cc [hunk 2]
```diff
@@ -0,0 +1,257 @@
...
+      // meters
+      _sub8_planner->as<ompl::geometric::RRTConnect>()->setRange(range);
+  }
+
+  // initialize alarms
+  std::string alarms_dir;
+  ros::param::get("alarms_dir", alarms_dir);
+  std::string pkg_path = ros::package::getPath("sub8_trajectory_generator");
+  char file_sep = '/';
+#ifdef _WIN32
```
Comment (jpanikulam) [inline gnc/sub8_trajectory_generator/src/tgen_manager.cc]: ?

File: gnc/sub8_trajectory_generator/test/tgen_manager_test.cc [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+  }
+
+  const ompl::base::StateSpacePtr& ss() const { return _test_space; }
+
+ private:
+  boost::shared_ptr<ros::NodeHandle> _node_handle;
+  ompl::base::StateSpacePtr _test_space;
+};
+
+TEST_F(TGenManagerTest, testSetProblemDefinition) {
```
Comment (jpanikulam) [inline gnc/sub8_trajectory_generator/test/tgen_manager_test.cc:45]: What's this testing?

File: gnc/sub8_trajectory_generator/test/tgen_integration.test [hunk 4]
```diff
@@ -0,0 +1,8 @@
+<launch>
+  <param name="simulate" value="True" type="bool"/>
+  <node pkg="sub8_videoray_m5_thruster" type="thruster_driver.py" name="thruster_driver"
+        args="--configuration_path=$(find sub8_videoray_m5_thruster)/config/calibration.json" />
+  <include file="$(find sub8_thruster_mapper)/launch/thruster_mapper.launch" />  
+  <include file="$(find sub8_trajectory_generator)/launch/trajectory_generator_params.launch" />
+  <test test-name="tgen_integration" name="tgen_integration_test" pkg="sub8_trajectory_generator" type="sub8_trajectory_generator_integration_tests" time-limit="60.0"/>
```
Comment (jpanikulam) [inline gnc/sub8_trajectory_generator/test/tgen_integration.test:7]: Do the integration tests verify that the tgen can take PathPlan service requests and yield a path?

**Comment by jpanikulam** [comment]
Misc comments for future:
- Try to squash sequential merges from master (db82812 and 5a50b44)
- 910a944 looks like a redundant merge?

Very pristine for a 2,000 line change commit!

File: gnc/sub8_trajectory_generator/src/tgen_manager.cc (hunk 2 above)
Comment (pemami4911) [inline gnc/sub8_trajectory_generator/src/tgen_manager.cc]: C++ alarms allow the user to specify alarms with JSON, by placing them in an "alarms" directory. I'm gonna move most of this code to the alarm helper class though in a new commit.

File: gnc/sub8_trajectory_generator/test/tgen_manager_test.cc (hunk 3 above)
Comment (pemami4911) [inline gnc/sub8_trajectory_generator/test/tgen_manager_test.cc:45]: This is just a test fixture for creating an instance of the TGenManager class for each test

File: gnc/sub8_trajectory_generator/test/tgen_integration.test (hunk 4 above)
Comment (pemami4911) [inline gnc/sub8_trajectory_generator/test/tgen_integration.test:7]: Currently no, will add one in a new commit

Comment (jpanikulam) [inline gnc/sub8_trajectory_generator/test/tgen_integration.test:7]: Ok - will pull on that.

File: gnc/sub8_trajectory_generator/src/tgen_manager.cc (hunk 2 above)
Comment (jpanikulam) [inline gnc/sub8_trajectory_generator/src/tgen_manager.cc]: Sorry - poor expression; Why ifdef _WIN32?

**Comment by jpanikulam** [comment]
Ready to pull once merge conflicts are resolved.

**Pull request merged**